operations/accountnotes.md

```diff
@@ -0,0 +1,341 @@
+
+### Response
+
+```javascript
+{
+    "AccountNotes": [
+        {
+            "Id": "a58ff7cb-77e3-495a-bd61-aecf00a3f19d",
+            "AccountId": "1df21f06-0cfc-4960-9c58-a3bf1261663e",
+            "Content": "Brother of the CEO",
+            "AccountType": "Customer",
+            "Classifications": ["FamilyRelations"],
+            "IsActive": true,
+            "CreatorProfile": {
+                "Discriminator": "Enterprise",
+                "EnterpriseProfile": {
+                    "ProfileId": "52d19c34-b0aa-4635-905d-1326fa8b8e13"
+                }
+            },
+            "UpdaterProfile": {
+                "Discriminator": "Integration",
+                "EnterpriseProfile": null
+            }
+        }
+    ],
+}
+```
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `AccountNotes` | array of [Account notes](#account-note) | required | Added account notes. |
+
+## Update account notes
+
+Updates information about the specified account notes.
+
+### Request
+
+`[PlatformAddress]/api/connector/v1/accountNotes/update`
+
+```javascript
+{
+    "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
+    "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
+    "Client": "Sample Client 1.0.0",
+    "AccountNoteUpdates": [
+        {
+            "AccountNoteId": "a58ff7cb-77e3-495a-bd61-aecf00a3f19d",
+            "Content": {
+                "Value": "The AC in the room doesn't work anymore"
+            },
+            "Classifications": {
+                "Maintenance": {
+                    "Value": true
+                },
+                "Housekeeping": {
+                    "Value": false
+                }
+            }
+        }
+    ]
+}
+```
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `ClientToken` | string | required | Token identifying the client application. |
+| `AccessToken` | string | required | Access token of the client application. |
+| `Client` | string | required | Name and version of the client application. |
+| `AccountNoteUpdates` | array of [Account note update parameters](#account-note-update-parameters) | required, max 1000 items | Account notes to be updated. |
+
+#### Account note update parameters
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `AccountNoteId` | string | required | Unique identifier of the account note. |
+| `Content` | [String update value](_objects.md#string-update-value) | optional, max length 1000 characters | Content of the account note \(or `null` if the content should not be updated\). |
+| `Classifications` | [Customer note classifications update parameters](#customer-note-classifications-update-parameters) or [Company note classifications update parameters](#company-note-classifications-update-parameters) | required | Classification of the account note |
```

Comment:
I missed this during the contract review for /update endpoint, but I believe I've mentioned it on /add review - it is not good that the deserialization depends on some implied value (here it is the type of account). Either it should have the same contract with different validation or an explicitly specified discriminator.

Reply:
I think that's been updated when we pass to the update objects with bool values for classifications. Not sure if it makes sense to do it like that now 🤔 We would be making all customer-related-classifications available to companies in the request. Is that something we really want?
We will be evaluating the account type and classification type if it helps 😅)

Reply:
The principle would be the same as in /add case - have a common list of classification values for both types of account in the contract, but restrict them using validation and also specify that in documentation. I'm not saying that it needs to be changed, but a possible way would be to have AccountNoteUpdateParameters like this:
```
{
     ...
     IEnumerable<EnumUpdateValue<AccountNoteClassification>> Classifications { get; set; }
}
```

where EnumUpdateValue<T> would be something along the lines of
```
{
      T Value { get; set; }
      bool IsSet { get; set; }
}
```

Reply:
This actually would be nice for update scenario as well. But at some point there's been an agreement for list of enum strings for `/add`, and boolean values for `/update`.
So for `/add` you say no need to change it. What's the decision for `/update`? Should we update it with one classification object?

Reply:
I don't think it's worth redoing, just leave it how it is.